### PR TITLE
BL-1514: Refactor additional_title_link

### PR DESCRIPTION
**Pull request by dkinzer**
*  Updates method to not render a link if only relation is present
*  Adds some tests to this helper method.

### CODE REVIEW

File: app/helpers/catalog_helper.rb [hunk 1]
```diff
@@ -370,24 +370,14 @@ def list_with_links(args)
...
-      end
-      link = fielded_search(linked_subfields, args[:field])
-
-      content_tag(:li, class: "list_items") do
-        if relation_to_work_prefix.present?
-          link.prepend("#{relation_to_work_prefix} ")
-        else
-          link
-        end
+    args[:document][args[:field]]
```
Comment (sensei100) [inline app/helpers/catalog_helper.rb:373]: @dkinzer Have they updated the requirements of this work?  The whole point of the previous work relating to this ticket is that they do not want the relation to be a part of the link.  I think the he only thing that should be considered a relation is 246 subfield i.

Comment (dkinzer) [reply app/helpers/catalog_helper.rb:373]: Could be they updated the requirements of this work. @ebtoner could we talk about this.  Just based on the expected links  provided in the issue, I assumed we wanted everything to be a link including relations which is what I did.

Comment (sensei100) [reply app/helpers/catalog_helper.rb:373]: @dkinzer I'm pretty sure the work for this needs to take place in cob_index.  That is where we are defining what should be a relation or title and that is where the misapplication is occurring.  The only things that should actually be labeled as relation are 246 subfield i.

Comment (dkinzer) [reply app/helpers/catalog_helper.rb:373]: Thanks, there's still  abug that happens when title is nil. Which will need to be fixed in this method.  I'll connect with @htomren about the details of the indexing.

**Review by sensei100 — APPROVED** [review]

**Pull request merged**